pre_commit/meta_hooks/check_useless_excludes.py

```diff
@@ -0,0 +1,41 @@
+import re
+import sys
+
+import pre_commit.constants as C
+from pre_commit.clientlib import load_config
+from pre_commit.git import get_all_files
+
+
+def exclude_matches_any(filenames, include, exclude):
+    include_re, exclude_re = re.compile(include), re.compile(exclude)
+    for filename in filenames:
+        if include_re.search(filename) and exclude_re.search(filename):
+            return True
+    return False
+
+
+def check_useless_excludes(config_file=None):
+    config = load_config(config_file or C.CONFIG_FILE)
+    files = get_all_files()
+    useless_excludes = False
+
+    exclude = config.get('exclude')
+    if exclude != '^$' and not exclude_matches_any(files, '', exclude):
+        print('The global exclude pattern does not match any files')
```

Comment:
Let's include `'{}'` in the error messages so we know *what* pattern failed to match